baybe/targets/numerical.py

```diff
@@ -1,40 +1,52 @@
 """Numerical targets."""
 
-import logging
+import warnings
 from functools import partial
-from typing import Any, Literal, Optional
+from typing import Any, Callable, Dict, Optional, Sequence, cast
 
+import numpy as np
 import pandas as pd
-from attr import define, field
+from attrs import define, field
+from numpy.typing import ArrayLike
 
 from baybe.targets.base import Target
-from baybe.utils import (
-    Interval,
-    SerialMixin,
-    bound_bell,
-    bound_linear,
-    bound_triangular,
-    convert_bounds,
+from baybe.targets.enum import TargetMode, TargetTransformation
+from baybe.targets.transforms import (
+    bell_transform,
+    linear_transform,
+    triangular_transform,
 )
+from baybe.utils import Interval, SerialMixin, convert_bounds
 
-_logger = logging.getLogger(__name__)
-
-
-# TODO: potentially introduce an abstract base class for the transforms
-#   -> this would remove the necessity to maintain the following dict
-_VALID_TRANSFORMS = {
-    "MAX": ["LINEAR"],
-    "MIN": ["LINEAR"],
-    "MATCH": ["TRIANGULAR", "BELL"],
+_VALID_TRANSFORMATIONS: Dict[TargetMode, Sequence[TargetTransformation]] = {
+    TargetMode.MAX: (TargetTransformation.LINEAR,),
+    TargetMode.MIN: (TargetTransformation.LINEAR,),
+    TargetMode.MATCH: (TargetTransformation.TRIANGULAR, TargetTransformation.BELL),
 }
+"""A mapping from target modes to allowed target transformations.
+If multiple transformations are allowed, the first entry is used as default option."""
+
+
+def _get_target_transformation(
+    mode: TargetMode, transformation: TargetTransformation
+) -> Callable[[ArrayLike, float, float], np.ndarray]:
+    """Provide the transform callable for the given target mode and transform type."""
+    if transformation is TargetTransformation.TRIANGULAR:
+        return triangular_transform
+    if transformation is TargetTransformation.BELL:
+        return bell_transform
+    if transformation is TargetTransformation.LINEAR:
+        if mode is TargetMode.MAX:
+            return partial(linear_transform, descending=False)
+        elif mode is TargetMode.MIN:
+            return partial(linear_transform, descending=True)
+    raise RuntimeError("This line should be impossible to reach.")
```

Comment:
should prob replace this by something more professional / informative

Reply:
Hm, yeah, not sure though what is the best solution here. BTW: we have this at several other place as well. Just for context: for typing reasons, you need to have "something" here that avoids that the function terminates without returning anything (because that would mess with the specified return type). So a simple `assert False` would effectively do the same job. We could also add an "informative" message like `Unknown mode {mode}`  but knowing that the message will never be shown since the line is unreachable. Thoughts?

Reply:
Perhaps the only "sensible" case to catch is when the caller does not enter a TargetMode in the first place... 🤔  So how about I change this into a `ValueError` and really turn the message into: `Unknown mode: {mode}`?

Reply:
its not `that` unreachable, assume someone meddles with a target object and overwrites the mode for whatever reason, they might end up here. So in that case I'd simply prefer if the message says unrecognized target mode etc, its much better than "This should be unreachable" because clearly it was reachable

Reply:
Ok, then I think we have the same situation in mind. Let me change it accordingly and then I'll merge. 👍🏼 
(And just FYI: people can't mess with the Target object, because I made it frozen 😎 ... at least not accidentally)

Reply:
Have a look now and resolve if you agree